crates/parser/src/grammar/generic_params.rs

```diff
@@ -144,6 +144,10 @@ fn type_bound(p: &mut Parser) -> bool {
             }
             if paths::is_use_path_start(p) {
                 types::path_type_(p, false);
+            } else if p.at(T![for]) {
+                // test question_for_type_trait_bound
+                // fn f<T>() where T: ?for<> Sized {}
+                types::for_type(p, false);
```

Comment:
Hmm, this allows `fn f<T>() where T: ~const for<> Sized {}`.  

One question: does our parser need to be 100% equivalent to rustc's? Or do we just need to make sure that those that can be parsed by rustc can be parsed by ra. For those syntaxes that can be parsed by ra but not by rustc, `flycheck` should be able to help us diagnose